addons/l10n_in/models/res_partner.py

```diff
@@ -28,6 +28,23 @@ class ResPartner(models.Model):
     )
 
     display_pan_warning = fields.Boolean(string="Display pan warning", compute="_compute_display_pan_warning")
+    l10n_in_gst_state_warning = fields.Char(compute="_compute_l10n_in_gst_state_warning")
+
+    @api.depends_context('company')
+    @api.depends('vat', 'state_id', 'country_id')
+    def _compute_l10n_in_gst_state_warning(self):
+        if self.env.company.country_code == "IN":
+            for partner in self.filtered(lambda p: p.check_vat_in(p.vat)):
+                if partner.vat[:2] == "99":
+                    partner.l10n_in_gst_state_warning = _("As per GSTN the country should be other than India, so it's recommended to update it.")
+                else:
+                    state_id = partner.env['res.country.state'].search([('l10n_in_tin', '=', partner.vat[:2])], limit=1)
```

Comment:
```suggestion
                    state_id = self.env['res.country.state'].search([('l10n_in_tin', '=', partner.vat[:2])], limit=1)
```
I guess you missed out suggestions by William